R/fetch_metal_pdb.R

```diff
@@ -0,0 +1,240 @@
+#' the atom can be specified here. E.g. "S" for sulfur. 
+#' @param columns optional character vector, if only specific columns should be retrieved these can be specified here based on the MetalPDB 
+#' \href{https://metalpdb.cerm.unifi.it/api_help}{website}. If nothing is supplied here, all possible columns will be retrieved. 
+#' @param show_progress logical, if true, a progress bar will be shown. Default is TRUE.
+#'
+#' @return A data frame that contains information about protein-metal binding sites.
+#' @import dplyr
+#' @import progress
+#' @import purrr
+#' @import tidyr
+#' @importFrom curl has_internet
+#' @importFrom magrittr %>%
+#' @export
+#'
+#' @examples
+#' \donttest{
+#' head(fetch_metal_pdb(id_value = c("P42345", "P00918")))
+#' 
+#' fetch_metal_pdb(id_type = "pdb", id_value = c("1g54"), metal = "Zn")
+#' }
+
+fetch_metal_pdb <- function(id_type = "uniprot",
+                            id_value,
+                            site_type = NULL,
+                            pfam = NULL,
+                            cath = NULL,
+                            scop = NULL,
+                            representative = NULL,
+                            metal = NULL,
+                            ligands = NULL,
+                            geometry = NULL,
+                            coordination = NULL,
+                            donors = NULL,
+                            columns = NULL,
+                            show_progress = TRUE) {
+  if (!requireNamespace("httr", quietly = TRUE)) {
+    stop("Package \"httr\" is needed for this function to work. Please install it.", call. = FALSE)
+  }
+  
+  if (!curl::has_internet()) {
+    message("No internet connection.")
+    return(invisible(NULL))
+  }
+  
+  . = NULL
+  # Make sure arguments are provided correctly
+  if (!missing(site_type)) {
+    if (length(site_type) > 1) {
+      stop('Please only provide one value for the "Site Type (Nuclearity)"')
+    }
+    site_type <- paste0(",site_type:", site_type)
+  }
+  
+  if (!missing(pfam)) {
+    if (length(pfam) > 1) {
+      stop('Please only provide one value for the "Pfam domain name"')
+    }
+    pfam <- paste0(",pfam:", pfam)
+  }
+  
+  if (!missing(cath)) {
+    if (length(cath) > 1) {
+      stop('Please only provide one value for the "Cath ID"')
+    }
+    cath <- paste0(",cath:", cath)
+  }
+  
+  if (!missing(scop)) {
+    if (length(scop) > 1) {
+      stop('Please only provide one value for the "SCOP ID"')
+    }
+    scop <- paste0(",scop:", scop)
+  }
+  
+  if (!missing(representative)) {
+    if (representative != "true") {
+      stop('Please only provide "true" for the "representative" argument')
+    }
+    representative <- paste0(",representative:", representative)
+  }
+  
+  if (!missing(metal)) {
+    if (length(metal) > 1) {
+      stop('Please only provide one value for the "Metal symbol"')
```

Comment:
should this be called "Metal symbol" ? Could you not call it "metal" argument? I think this would make the message a bit more informative.